persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java

```diff
@@ -68,228 +72,257 @@
+      }
+
       if (configuration.rawValues() && iceFactory == null) {
          iceFactory = ctx.getCache().getAdvancedCache().getComponentRegistry().getComponent(InternalEntryFactory.class);
       }
+
+      // Make sure threads are marked as non blocking if user didn't specify
+      configuration.properties().putIfAbsent("blocking", "false");
+
+      ConfigurationBuilder builder = buildRemoteConfiguration(configuration, marshaller);
+
+      return blockingManager.supplyBlocking(() -> {
+         remoteCacheManager = new RemoteCacheManager(builder.build());
+
+         if (configuration.remoteCacheName().isEmpty())
+            remoteCache = (InternalRemoteCache<Object, Object>) remoteCacheManager.getCache();
+         else
+            remoteCache = (InternalRemoteCache<Object, Object>) remoteCacheManager.getCache(configuration.remoteCacheName());
+
+         return remoteCache.ping();
+      }, "RemoteStore-start")
+            .thenCompose(Function.identity())
+            .thenAccept(pingResponse -> {
+               String cacheName = ctx.getCache().getName();
+               StorageConfigurationManager storageConfigurationManager = ctx.getCache().getAdvancedCache().getComponentRegistry()
+                     .getComponent(StorageConfigurationManager.class);
+               MediaType localKeyStorageType = storageConfigurationManager.getKeyStorageMediaType();
+               MediaType serverKeyStorageType = pingResponse.getKeyMediaType();
+
+               MediaType localValueStorageType = storageConfigurationManager.getValueStorageMediaType();
+               MediaType serverValueStorageType = pingResponse.getValueMediaType();
+
+               DataFormat.Builder dataFormatBuilder = DataFormat.builder().from(remoteCache.getDataFormat());
+               Integer numSegments = remoteCache.getCacheTopologyInfo().getNumSegments();
+               boolean segmentsMatch;
+               if (numSegments == null) {
+                  log.debugf("Remote Store for cache %s cannot support segmentation as the number of segments was not found from the remote cache", cacheName);
+                  segmentsMatch = false;
+               } else {
+                  segmentsMatch = numSegments == segmentCount;
+                  if (segmentsMatch) {
+                     log.debugf("Remote Store for cache %s can support segmentation as the number of segments matched the remote cache", cacheName);
+                  } else {
+                     log.debugf("Remote Store for cache %s cannot support segmentation as the number of segments %d do not match the remote cache %d",
+                           cacheName, segmentCount, numSegments);
+                  }
+               }
+               if (!segmentsMatch && configuration.segmented()) {
+                  throw log.segmentationRequiresEqualSegments(segmentCount, numSegments);
+               }
+               if ((supportsSegmentation = localKeyStorageType.equals(serverKeyStorageType))) {
```

Comment:
unnecessary additional brackets

Reply:
Technically, they are needed as I am assigning a variable in there. Dan doesn't like these and usually makes me move them to a different line :D